### PR TITLE
bump gerrit to 2.15.3

### PR DESCRIPTION
**Pull request by phedoreanu**
This PR contains small improvements that help with the deployment and administration of a Gerrit server.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,15 +6,15 @@ MAINTAINER zsx <thinkernel@gmail.com>
...
+ENV GERRIT_VERSION 2.15.3
 ENV GERRIT_USER gerrit2
 ENV GERRIT_INIT_ARGS ""
 
 # Add our user and group first to make sure their IDs get assigned consistently, regardless of whatever dependencies get added
 RUN adduser -D -h "${GERRIT_HOME}" -g "Gerrit User" -s /sbin/nologin "${GERRIT_USER}"
 
 RUN set -x \
-    && apk add --update --no-cache git openssh openssl bash perl perl-cgi git-gitweb curl su-exec procmail
+    && apk add --update --no-cache git openssh-client openssl bash perl perl-cgi git-gitweb curl su-exec procmail
```
Comment (thinkernel) [inline Dockerfile]: Are you sure openssh-client is good enough? I added this years ago. I can't remember why I used openssh instead of openssh-client. :P So I hope you've already test this on your environment.

File: Dockerfile [hunk 2]
```diff
@@ -71,6 +71,4 @@ VOLUME $GERRIT_SITE
 
 ENTRYPOINT ["/gerrit-entrypoint.sh"]
 
-EXPOSE 8080 29418
-
```
Comment (thinkernel) [inline Dockerfile]: Any idea that we should remove this one?

File: Dockerfile (hunk 1 above)
Comment (phedoreanu) [reply Dockerfile]: `openssh` is a group, it installs both `openssh-client` and `openssh-server` and I figured Gerrit doesn't need the `server`. I have tested it, yes and everything works as expected:
```
2018-07-25 10:54:52,600 +0000] 053a33df test_user a/1000000 LOGIN FROM 172.1.19.78
[2018-07-25 10:54:53,313 +0000] 053a33df test_user a/1000000 git-receive-pack./dm 72ms 487ms 0
[2018-07-25 10:54:53,340 +0000] 053a33df test_user a/1000000 LOGOUT
[2018-07-25 10:54:57,433 +0000] 65668f08 test_user a/1000000 LOGIN FROM 172.1.19.78
[2018-07-25 10:54:57,627 +0000] 65668f08 test_user a/1000000 git-upload-pack./dm 7ms 88ms 0
[2018-07-25 10:54:57,653 +0000] 65668f08 test_user a/1000000 LOGOUT
[2018-07-25 10:54:59,684 +0000] e5511f21 test_user a/1000000 LOGIN FROM 172.1.19.78
[2018-07-25 10:54:59,882 +0000] e5511f21 test_user a/1000000 git-receive-pack./dm 3ms 89ms 0
[2018-07-25 10:54:59,953 +0000] e5511f21 test_user a/1000000 LOGOUT
```

File: Dockerfile (hunk 2 above)
Comment (phedoreanu) [reply Dockerfile]: I have added `EXPOSE` back in, but it isn't required - [expose docs](https://docs.docker.com/engine/reference/builder/#expose).

> The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container...

File: gerrit-entrypoint.sh [hunk 3]
```diff
@@ -83,13 +83,18 @@ EOF
 done
 
   #Customize gerrit.config
+  [ -z "${MAKE_FIRST_USER_ADMIN}" ] || set_gerrit_config capability.makeFirstUserAdmin "${MAKE_FIRST_USER_ADMIN}"
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: This is a 2.15.x only config option.

File: gerrit-entrypoint.sh [hunk 4]
```diff
@@ -83,13 +83,18 @@ EOF
...
+  [ -z "${ENABLE_SIGNED_PUSH}" ] || set_gerrit_config receive.enableSignedPush "${ENABLE_SIGNED_PUSH}"
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: This option is existed in both 2.14.x and 2.15.x.
I'd like to merge this kind of options back to 2.14.x branch. So a separated commit would be welcome.

File: gerrit-entrypoint.sh [hunk 5]
```diff
@@ -83,13 +83,18 @@ EOF
...
+  [ -z "${DOWNLOAD_SCHEME}" ] || set_gerrit_config download.scheme "${DOWNLOAD_SCHEME}"
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: The download.scheme options can be set multi-times. This will make it only support one scheme.

File: gerrit-entrypoint.sh [hunk 6]
```diff
@@ -83,13 +83,18 @@ EOF
...
 
   #Section gerrit
   [ -z "${WEBURL}" ] || set_gerrit_config gerrit.canonicalWebUrl "${WEBURL}"
   [ -z "${GITHTTPURL}" ] || set_gerrit_config gerrit.gitHttpUrl "${GITHTTPURL}"
 
   #Section sshd
-  [ -z "${LISTEN_ADDR}" ] || set_gerrit_config sshd.listenAddress "${LISTEN_ADDR}"
+  [ -z "${SSHD_LISTEN_ADDR}" ] || set_gerrit_config sshd.listenAddress "${SSHD_LISTEN_ADDR}"
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: I agree that SSHD_LISTEN_ADDR is better than LISTEN_ADDR. However I'd like keep LISTEN_ADDR works in order to not break those exist containers to upgrade to the new image.

File: gerrit-entrypoint.sh [hunk 7]
```diff
@@ -83,13 +83,18 @@ EOF
...
+  [ -z "${SSHD_ADVERTISE_ADDR}" ] || set_gerrit_config sshd.advertisedAddress "${SSHD_ADVERTISE_ADDR}"
+  [ -z "${SSHD_ENABLE_COMPRESSION}" ] || set_gerrit_config sshd.enableCompression "${SSHD_ENABLE_COMPRESSION}"
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: This two options are existed in both 2.14.x and 2.15.x.
I'd like to merge them back to 2.14.x branch. So a separated commit would be welcome.

File: gerrit-entrypoint.sh [hunk 8]
```diff
@@ -83,6 +83,8 @@ EOF
 done
 
   #Customize gerrit.config
+  set_gerrit_config repository."*".ownerGroup "Registered Users"
+
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: To provide one way to do the setup is not a bad idea. However, I'd like to keep this option more open for those who don't want "Registered Users" as the default owner of new projects.

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    - postgresdb
+    restart: unless-stopped
+  postgresdb:
+    image: postgres:alpine
+    expose:
+    - 5432
+    environment:
+      POSTGRES_USER: gerrit2
+      POSTGRES_PASSWORD: gerrit
+      POSTGRES_DB: reviewdb
```
Comment (thinkernel) [inline docker-compose.yml]: In general, I'd like to keep this project as simple as possible. That means this project is only about building a docker container. docker-compose or k8s-deployments should be separated to other projects.

File: gerrit-entrypoint.sh [hunk 10]
```diff
@@ -85,9 +85,15 @@ done
   #Customize gerrit.config
   set_gerrit_config repository."*".ownerGroup "Registered Users"
 
+  if [ -n "${DOWNLOAD_SCHEMES}" ]; then
+    set_gerrit_config --remove-section download || true
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: Is there a way that only remove those download.scheme, instead of remove the whole download section?

File: gerrit-entrypoint.sh [hunk 11]
```diff
@@ -54,6 +54,34 @@ if [ "$1" = "/gerrit-start.sh" ]; then
...
+[remote "$r"]
+	url = $REMOTE
+	push = +refs/tags/*:refs/tags/*
+	push = +refs/heads/*:refs/heads/*
+	mirror = $MIRROR
+	projects = $PROJECTS
+	replicateOnStartup = $REPLICATE_ON_STARTUP
+EOF
+done
+
```
Comment (thinkernel) [inline gerrit-entrypoint.sh]: To support replication.config is a great feature. However I'd like to separate and review this feature in another PR.

**Comment by phedoreanu** [comment]
Hi @thinkernel 

Thanks for the feedback! Commits squashed with only commits of the version bump.

File: docker-compose.yml (hunk 9 above)
Comment (phedoreanu) [reply docker-compose.yml]: I understand, but it serves more for documentation on how to run **this** image. Maybe an `example` folder would more intuitive?

File: gerrit-entrypoint.sh (hunk 10 above)
Comment (phedoreanu) [reply gerrit-entrypoint.sh]: I'll try with `--unset-all download.scheme`.
